numba/cuda/tests/cudapy/test_powi.py

```diff
@@ -45,6 +71,54 @@ def test_powi_binop(self):
         kernel[1, A.shape](A, power, Aout)
         self.assertTrue(np.allclose(Aout, A ** power))
 
+    # Relative tolerance kwarg is provided because 1.0e-7 (the default for
+    # assert_allclose) is a bit tight for single precision.
```

Comment:
Does this end up with 64bit floating point, by default, using a far slacker tolerance than would normally be expected of it?

Reply:
The default here is the same as the default for `assert_allclose`, which is `1.0e-7` - it's only for single precision that it's overridden with `3.0e-7`. So if I've understood your question correctly, I think the answer is "no".

Reply:
Thanks for the explanation. Numba's CPU impl tends to use `self.assertPreciseEqual` for testing fundamental maths functions, should the CUDA target do similar?

Reply:
Maybe. At the moment most maths functions in the CUDA tests are tested to the rather loose relative tolerance of 1.0e-5. I think it would take some time going through them and determining to what extent there are differences between it and the CPU target / libc implementations to fix up in general, but I'll check if `assertPreciseEqual` works here.